### PR TITLE
fix(backends): ensure that returned date results are actually proper date values

### LINKED ISSUE #7299: bug: DateScalar repr says it's a Timestamp
### What happened?

`ibis.date("2000-01-01")` presents itself to `Timestamp('2000-01-01 00:00:00')`. This really confused me for a while.

This is different from if you have a column, which advertises itself as a date:
```
ibis.memtable({"date": ["2020-01-01", "2020-01-02", "2020-01-03"]}).date.cast("date")
┏━━━━━━━━━━━━━━━━━━┓
┃ Cast(date, date) ┃
┡━━━━━━━━━━━━━━━━━━┩
│ date             │
├──────────────────┤
│ 2020-01-01       │
│ 2020-01-02       │
│ 2020-01-03       │
└──────────────────┘
```
I would expect the DateScalar to look like the column, perhaps `Date('2000-01-01')`

### What version of ibis are you using?

7.0.0

### What backend(s) are you using, if any?

duckdb

### Relevant log output

_No response_

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

### PR DESCRIPTION
**Pull request by cpcloud**
Changes date types to actually be dates in all cases. Closes #7299.

### CODE REVIEW

**Comment by kszucs** [comment]
@cpcloud can you squash?

**Comment by cpcloud** [comment]
@kszucs Done!

File: ibis/backends/pandas/execution/temporal.py [hunk 1]
```diff
@@ -65,9 +65,19 @@ def execute_extract_microsecond_series(op, data, **kwargs):
     return data.dt.microsecond.astype(np.int32)
 
 
-@execute_node.register(ops.ExtractEpochSeconds, (datetime.datetime, pd.Series))
-def execute_epoch_seconds(op, data, **kwargs):
-    return data.astype("datetime64[s]").astype("int64").astype("int32")
+@execute_node.register(ops.ExtractEpochSeconds, pd.Series)
+def execute_epoch_seconds_series(op, data, **kwargs):
+    return (
```
Comment (kszucs) [inline ibis/backends/pandas/execution/temporal.py:70]: One of my favourite casts :)

File: ibis/expr/types/generic.py [hunk 2]
```diff
@@ -1225,6 +1225,9 @@ def __pyarrow_result__(
         return data_mapper.convert_scalar(table[0][0], self.type())
 
     def __pandas_result__(self, df: pd.DataFrame) -> Any:
+        from ibis.formats.pandas import PandasData
+
+        df = PandasData.convert_table(df, ibis.schema({self.get_name(): self.type()}))
```
Comment (kszucs) [inline ibis/expr/types/generic.py:1230]: There should be a `PandasData.convert_scalar()` method, I added that in one of my reverted commits, let me look for it.

Comment (kszucs) [reply ibis/expr/types/generic.py:1230]: Here https://github.com/ibis-project/ibis/pull/7580/commits/ac0282c282459e22414de2ef96d02d8617b54751#diff-c3915f2476dac4d27022c405fd09c019b590114ef799dbddb3aab3a0c7c4b177L104-L108

Comment (kszucs) [reply ibis/expr/types/generic.py:1230]: Could you please add that to the `PandasData` mapper?

Comment (cpcloud) [reply ibis/expr/types/generic.py:1230]: Yep!

Comment (cpcloud) [reply ibis/expr/types/generic.py]: Done!

**Comment by cpcloud** [comment]
Cloud backends look good:

```
…/ibis on  date-scalar-repr is 📦 v7.1.0 via 🐍 v3.10.13 via ❄️  impure (ibis-3.10.13-env)
❯ pytest -m 'bigquery or snowflake' -n 8 --dist loadgroup --snapshot-update -q
bringing up nodes...
ssssssssssssssssssssss..xxx.xxxxx.xxxx.xxx.x.xxxx....xx....xxx..xxxxxxx.x.x.xxxx......xx...x.xx.x.xxx.xx....x........xxx........x...........x.x.......xx..x....x.....x..xx.......x.x..x.x....x [  6%]
xx.....x......x...x.x.x....x..x.x.xxx...x..x...x....xx.x.xx.xx.....x...xx..xxx.x.x..x..xxxx....x....xx.x...xxx.x.x....xx............x....xxxx.x.x..xx.x.xx...x..x..x.......x..............x... [ 12%]
.x.....xx.x...x..x.............x.........x..........x.................x................x...xx.x......x....x........xx.......xx..xx..x....x.x...xx....x..x.........x.....x...x.............x... [ 18%]
....x....x...x..x...xx......x....x..x....x.....x.......x...x.......x..x............x......x..x........x..x.x...x.............................................................................. [ 24%]
......................................................x.........................x............................................................................x..s..............x.......x.x.xx. [ 30%]
..x.....x...xx....x..x.xxxxx.xx...xxxxx..xxx....x......x....x....s.............x....sx...x........s..x...............x..................x....xx.......x...x.xx.xxxxx.x.xxxxxxxxxx.xxxxxxxxxx.x [ 36%]
...x..x.....xx...x........x..xx.x.xx.x......x..x........xx......x.xx............xx.............x......x........x.............xx.........x..xx...........x.x.......................x........... [ 42%]
.......x......x........x...x..x..x.....x....x...........x.x..................x.x....x...xx.....x..............x...x.x..x......x...........x.....x.x..x..x.....xx......x..........x.........x.. [ 49%]
..........x..............x....................x...............x.................................s...............................x................x..x..x.xx.x..........x...x.x.....x...xx...x. [ 55%]
...xx.x.x....xx..........x...x....x....x..x..x......................x..........x............................x......x..................................x....................................... [ 61%]
.........................................................................x................x.........x...........x.....x......................x..............x.......x...x.xx.................. [ 67%]
..x.....s.........x.....x..................x....x.x............x.x..xx..x.xxxxxxxxx.x..x..xx..x..x..........x....................x......x............x.x..........x.x..x.......xx....x...x.x.. [ 73%]
.....x....x....x.....x...........................................x.x...................x..x...............xx.....x.....x...............................x....x...........x......x..........x.x. [ 79%]
....................x....xx...x..s.....................................x.x.................................................................................................................... [ 85%]
....................................x...................x.......................................x......x.x......x...........x.......x.............xx.....x....x.....x......................... [ 91%]
................................s.......................................................................................................................................x.x.x.xxx............. [ 98%]
...............x..........................................                                                                                                                                     [100%]
2618 passed, 30 skipped, 450 xfailed in 798.61s (0:13:18)
```

**Pull request merged**